correlation_context/HTTP_HEADER_FORMAT.md

```diff
@@ -1,43 +1,71 @@
 
 *See [rationale document](HTTP_HEADER_FORMAT_RATIONALE.md) for details of decisions made for this format.*
 
-# Format
 
-## Header name
+# Header Name
+
+Header name: `Correlation-Context`
+
+@TODO: Agree on the header name
+@TODO: Add more information on casing once we agree on the final header name
 
-`Correlation-Context`
 
-Multiple correlation context headers are allowed. Values can be combined in a single header according to the [RFC 7230](https://tools.ietf.org/html/rfc7230#page-24).
+# Header Content
 
-## Header value
+This section uses the Augmented Backus-Naur Form (ABNF) notation of [[!RFC5234]], including the DIGIT rule in <a data-cite='!RFC5234#appendix-B.1'>appendix B.1 for RFC5234</a>. It also includes the `OWS` rule from <a data-cite='!RFC7230#whitespace'>RFC7230 section 3.2.3</a>.
 
-`name1=value1[;properties1],name2=value2[;properties2]`
+## Definition
+
+```
+list        = list-member 0*179( OWS "," OWS list-member )
+list-member = key OWS "=" OWS value *( OWS ";" OWS property )
+property    = key OWS "=" OWS value
+property    = key OWS
+key         = <token, defined in [[RFC2616], Section 2.2](https://tools.ietf.org/html/rfc2616#section-2.2)>
+value       = %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
+              ; US-ASCII characters excluding CTLs,
+              ; whitespace DQUOTE, comma, semicolon,
+              ; and backslash
+OWS         = <Optional white space, as defined in [[RFC7230], Section 3.2.3.](https://tools.ietf.org/html/rfc7230#section-3.2.3)>
+```
 
-**Limits:**
+## Limits
 1. Maximum number of name-value pairs: `180`.
 2. Maximum number of bytes per a single name-value pair: `4096`.
 3. Maximum total length of all name-value pairs: `8192`.
 
-## Name format
+## Example
+`key1=value1;property1;property2, key2 = value2; property3, key3=value3;propertyKey=propertyValue`
+
+### list
+List of key-value pairs with optional properties attached.
+It can not be guaranteed that keys are unique.
+
+Consumers MUST be able to handle duplicate keys while producers SHOULD try to dedupe the list.
 
-Url encoded string. Spaces are allows before and after the name. Header with the trimmed name and with spaces before and after name MUST be considered identical.
 
-## Value format
+### key
+ASCII string according to the `token` format, defined in [[RFC2616], Section 2.2](https://tools.ietf.org/html/rfc2616#section-2.2). Leading and trailing OWS is allowed but MUST be trimmed when converting the header into a data structure.
 
-Value starts after equal sign and ends with the special character `;`, separator `,` or end of string. Value represents a url encoded string and case sensitive. Spaces are allowed in the beginning and the end of the value. Value with spaces before and after MUST be considered identical to the trimmed value. 
+### value
 
-## Properties
+The value contains an URL encoded UTF-8 string. Leading and trailing OWS is allowed but MUST be trimmed when converting the header into a data structure.
```

Comment:
> URL encoded ... string

I though we agreed not to require URL encoding, leaving it to producers/consumers of a given key to agree on representation that satisfies the protocol restrictions.

Reply:
We were leaning that way when you disconnected, but after talking about it more it seems there is not strong agreement either way. Mostly the reason for the encoding is that the values will somehow have to be encoded anyway so prescribing an encoding scheme makes it easier for e.g. APM vendors to read and display to users keys they may not "know" about.

Reply:
nvm

Reply:
Fair enough, I don't have a strong objection. It might be a bit weird when used with non-HTTP transports, but then we still do need to escape the commas, so url encoding is ok.